becquerel/core/fitting.py

```diff
@@ -0,0 +1,859 @@
+from __future__ import print_function
+import warnings
+import inspect
+import numpy as np
+import pandas as pd
+import scipy.special
+from lmfit.model import Model
+import matplotlib.pyplot as plt
+from matplotlib.gridspec import GridSpec
+from matplotlib.font_manager import FontProperties
+from .utils import isstring
+
+FWHM_SIG_RATIO = np.sqrt(8*np.log(2))  # 2.35482
+SQRT_TWO = np.sqrt(2) #1.414213562
+
+class FittingError(Exception):
+    """Exception raised by Fitters."""
+    pass
+
+
+class FittingWarning(UserWarning):
+    """Warning raised by Fitters."""
+
+
+# -----------------------------------------------------------------------------
+# Base functions
+# TODO: Should these be replaced with lmfit.models???
+# -----------------------------------------------------------------------------
+
+
+def constant(x, c):
+    return np.ones_like(x) * c
+
+
+def line(x, m, b):
+    return m * x + b
+
+
+def gauss(x, amp, mu, sigma):
+    return amp / sigma / np.sqrt(2. * np.pi) * \
+        np.exp(-(x - mu) ** 2. / (2. * sigma ** 2.))
+
+
+def erf(x, amp, mu, sigma):
+    return amp * 0.5 * (1. - scipy.special.erf((x - mu) / (SQRT_TWO * sigma)))
+
+
+def exp(x, amp, lam):
+    return amp * np.exp(x / lam)
+
+
+def gausserf(x, ampgauss, amperf, mu, sigma):
+    return gauss(x, ampgauss, mu, sigma) + erf(x, amperf, mu, sigma)
+
+
+def expgauss(x, amp=1, mu=0, sigma=1.0, gamma=1.0):
+    gss = gamma*sigma*sigma
+    arg1 = gamma*(mu + gss/2.0 - x)
+    arg2 = (mu + gss - x)/(SQRT_TWO*sigma)
+    return amp*(gamma/2) * np.exp(arg1) * scipy.special.erfc(arg2)
+
+# -----------------------------------------------------------------------------
+# Fitting models
+# -----------------------------------------------------------------------------
+
+
+# Helper functions for guessing
+def _xy_right(y, x=None, num=4):
+    if x is not None:
+        return np.mean(x[-num:]), np.mean(y[-num:])
+    return len(y)*0.5, np.mean(y[-num:])
+
+
+def _xy_left(y, x=None, num=4):
+    if x is not None:
+        return np.mean(x[:num]), np.mean(y[:num])
+    return len(y)*0.5, np.mean(y[:num])
+
+
+class ConstantModel(Model):
+
+    def __init__(self, *args, **kwargs):
+        super(ConstantModel, self).__init__(constant, *args, **kwargs)
+        # TODO: remove this min setting?
+        self.set_param_hint('{}c'.format(self.prefix), min=0.)
+
+    def guess(self, y, x=None, dx=None, num=2):
+        if dx is None:
+            dx = np.ones_like(x)
+        c = (y[-1]/dx[-1] + y[0]/dx[0]) * 0.5
+        return [
+            ('{}c'.format(self.prefix), 'value', c),
+        ]
+
+
+class LineModel(Model):
+
+    def __init__(self, *args, **kwargs):
+        super(LineModel, self).__init__(line, *args, **kwargs)
+
+    def guess(self, y, x=None, dx=None, num=2):
+        _, b = _xy_left(y, x=x, num=num)
```

Comment:
This `b` is never used

Reply:
Then this func call this is not needed right?

Reply:
Correct, I'll delete it